pkg/v2/tkr/controller/tkr-source/compatibility/reconciler.go

```diff
@@ -166,24 +191,56 @@ func (c *Compatibility) getManagementClusterVersion(ctx context.Context) (string
 		}
 	}
 
-	return "", errors.New("failed to get management cluster info")
+	c.Log.Info("Could not find the Cluster resource with needed metadata",
+		"label", constants.ManagementClusterRoleLabel, "annotation", constants.TKGVersionKey)
+
+	return "", nil
 }
 
 func (c *Compatibility) compatibilityMetadata(ctx context.Context) (*types.CompatibilityMetadata, error) {
 	cm := &corev1.ConfigMap{}
 	cmObjectKey := client.ObjectKey{Namespace: c.Config.TKRNamespace, Name: constants.BOMMetadataConfigMapName}
 	if err := c.Client.Get(ctx, cmObjectKey, cm); err != nil {
+		err = kerrors.FilterOut(err, apierrors.IsNotFound)
 		return nil, err
 	}
 
 	metadataContent, ok := cm.BinaryData[constants.BOMMetadataCompatibilityKey]
 	if !ok {
-		return nil, errors.New("compatibility key not found in bom-metadata ConfigMap")
+		c.Log.Error(errors.New("compatibility key not found in bom-metadata ConfigMap"), "This.")
```

Comment:
nit: What is the purpose of "This." ?

Reply:
That's the log message :) It is pointing that the subject of the log entry is the error that is being logged.